scripthost/src/test/kotlin/com/commonwealthrobotics/bowlerkernel/scripthost/ScriptHostTest.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * This file is part of bowler-kernel.
+ *
+ * bowler-kernel is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * bowler-kernel is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with bowler-kernel.  If not, see <https://www.gnu.org/licenses/>.
+ */
+package com.commonwealthrobotics.bowlerkernel.scripthost
+
+import com.commonwealthrobotics.bowlerkernel.protoutil.sessionClientMessage
+import com.commonwealthrobotics.bowlerkernel.testutil.KoinTestFixture
+import io.kotest.assertions.throwables.shouldThrow
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.Test
+import org.koin.dsl.module
+
+internal class ScriptHostTest : KoinTestFixture() {
+
+    @Test
+    fun `starting multiple concurrent sessions is an error`() {
+        testKoin(module {})
+        val host = ScriptHost(CoroutineScope(Dispatchers.Default))
+        host.session(flowOf())
+        shouldThrow<IllegalStateException> {
+            host.session(flowOf())
+        }
```

Comment:
This looks like a race condition; because the first session is based on an empty flow, it *could* finish entirely before `shouldThrow` starts.

Reply:
> it _could_ finish entirely before `shouldThrow` starts

It's never started because the created session's `session` property is never collected, so there is no race.